lib/httpMock.js

```diff
@@ -15,15 +15,13 @@ function mockTemplate() {
 
         $provide.decorator('$http', ['$delegate', '$q', '$injector', function($http, $q, $injector) {
 
-        var interceptors = $httpProvider.interceptors;
+        var interceptors = [];
 
-        function getInterceptor(interceptorExpression) {
-            if (angular.isString(interceptorExpression)) {
-                return $injector.get(interceptorExpression);
-            } else {
-                return $injector.invoke(interceptorExpression);
-            }
-        }
+        angular.forEach($httpProvider.interceptors, function(interceptorExpression) {
```

Comment:
Regarding added interceptors: This case should only happen if the $httpProvider is modified after the mock() call in a jasmine test scenario. The $http service can not modify interceptors. I think the normal case is to mock the $http service at the very and of a test scenario in a beforeEach() block. Therefore the mock is the last delegate of the $http service, isn't it?

And than this PR should be safe.